xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/extension/extension.js

```diff
@@ -495,6 +495,12 @@ XWiki.ExtensionBehaviour = Class.create({
       parameters : formData,
       onCreate : function() {
         dependency.removeClassName('extension-item-unknown').addClassName('extension-item-loading');
+        // Remove the unknown icon.
```

Comment:
This is not really clear why we need to do all this.
Also, I don't know this code much but why do you need to manipulate Velocity code from javascript?

Reply:
> Also, I don't know this code much but why do you need to manipulate Velocity code from javascript?

We could make it work with smart CSS relying on logic classes to hide and show icons. I tried to keep the changes to a minimum to avoid making this more complex. Keeping things simple means also that it's less likely to break things in a usecase/customization unknown to me. 

This code is quite old and large. A complete overhaul would be the best solution quality wise, but it's beyond the goal of this PR.

> This is not really clear why we need to do all this.

Before, the icon would swap directly when we swapped the classes. Here, I added a few lines to swap the icon DOM elements at the same time as they did before.

Reply:
TODO: At least add a comment to explain the HTML manipulations. Should probably add CSS logic to avoid this  HTML manipulations. Do leave things like this if extensive testing show that it's not possible to have a simple CSS logic in place...